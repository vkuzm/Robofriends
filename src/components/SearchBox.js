import React from 'react';

class SearchBox extends React.Component {
	shouldComponentUpdate() {
		return false;
	}

 	render() {
		const { searchChange } = this.props;

		return (
			<div className='pa2'>
		  	<input
		  		id="searchBox"
		  		className='pa3 ba b--green bg-lightest-blue'
		    	type='search'
		    	placeholder='Search robots'
		    	onChange={searchChange}
		   	/>
			</div>
		);
	}
}

export default SearchBox;