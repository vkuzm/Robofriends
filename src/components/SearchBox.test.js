import React from 'react';
import SearchBox from './SearchBox';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

it('expect to render SearchBox correctly', () => {
	const wrapperSnapshot = toJson(shallow(<SearchBox />));
  expect(wrapperSnapshot).toMatchSnapshot();
});

it('correctly searchChange', () => {
  const onSearchChange = function() {
    return true;
  }
  const wrapper = shallow(<SearchBox searchChange={onSearchChange} />);
  
  //wrapper.find('[id="searchBox"]');
  //wrapper.find('[id="searchBox"]').simulate('change');
  //expect(wrapper.find('[id="searchBox"]').value()).toEqual('test');
});