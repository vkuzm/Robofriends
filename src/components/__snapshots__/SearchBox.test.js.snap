// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render SearchBox correctly 1`] = `
<div
  className="pa2"
>
  <input
    className="pa3 ba b--green bg-lightest-blue"
    id="searchBox"
    placeholder="Search robots"
    type="search"
  />
</div>
`;
